dsl/camel-yaml-dsl/camel-yaml-dsl/src/main/java/org/apache/camel/dsl/yaml/YamlRoutesBuilderLoader.java

```diff
@@ -57,39 +61,46 @@ protected void doBuild() throws Exception {
         super.doBuild();
 
         this.settings = LoadSettings.builder().build();
-        this.constructor = new YamlDeserializationContext(settings);
-        this.constructor.setCamelContext(getCamelContext());
-        this.constructor.addResolvers(new CustomResolver());
-        this.constructor.addResolvers(new ModelDeserializersResolver());
-        this.constructor.addResolvers(new EndpointProducerDeserializersResolver());
+        this.deserializationContext = new YamlDeserializationContext(settings);
+        this.deserializationContext.setCamelContext(getCamelContext());
+        this.deserializationContext.addResolvers(new CustomResolver());
+        this.deserializationContext.addResolvers(new ModelDeserializersResolver());
+        this.deserializationContext.addResolvers(new EndpointProducerDeserializersResolver());
+
+        final Map<String, String> options = getCamelContext().getGlobalOptions();
```

Comment:
I wonder if the mode should be set later, eg in doInit or something, as potentially the end user may configure the mode on camel context global options at some point later, than this class may be created.

Reply:
ah yes, good point, so I'll move that to doStart so it will be always a runtime thing

Reply:
done